code/modules/research/design_database.dm

```diff
@@ -0,0 +1,182 @@
+/obj/machinery/design_database
+	name = "fabricator design database"
+	icon = 'icons/obj/machines/server.dmi'
+	icon_state = "server-off"
+	density = TRUE
+	anchored = TRUE
+
+	var/local_network_tag
+	var/list/tech_levels = list(
+		TECH_MATERIAL =    0,
+		TECH_ENGINEERING = 0,
+		TECH_PHORON =      0,
+		TECH_POWER =       0,
+		TECH_BLUESPACE =   0,
+		TECH_BIO =         0,
+		TECH_COMBAT =      0,
+		TECH_MAGNET =      0,
+		TECH_DATA =        0,
+		TECH_ESOTERIC =    0
+	)
+
+	var/need_disk_operation = FALSE
+	var/obj/item/disk/tech_disk/disk
+	var/sync_policy = SYNC_PULL_NETWORK|SYNC_PULL_DISK
+
+/obj/machinery/design_database/CanUseTopic(var/user)
+	if(istype(user, /mob/living/silicon) || (Adjacent(user)))
+		return STATUS_INTERACTIVE
```

Comment:
This is not a desirable override. If there's something in particular you're trying to accomplish with it, ask me.